1-starter-app/webapp/helper.py

```diff
@@ -0,0 +1,92 @@
+import json
+import os
+
+from azure.data.tables import TableServiceClient
+from dotenv import load_dotenv
+from flask import request
+
+load_dotenv(verbose=True)
+
+
+class TableServiceHelper:
+
+    def __init__(self, table_name=None, conn_str=None):
+        self.table_name = table_name if table_name else os.getenv("table_name")
+        self.conn_str = conn_str if conn_str else os.getenv("conn_str")
+
+    @property
+    def table_client(self):
+        pass #TODO
+
+    def query_entity(self, params):
+        return [] #TODO
+
+    def delete_entity(self):
+        pass #TODO
+
+    def insert_entity(self):
+        pass #TODO
+
+    def upsert_entity(self):
+        pass #TODO
+
+    def update_entity(self):
+        pass #TODO
+
+    def submit_transaction(self, operations):
+        pass #TODO
+
+    def insert_sample_data(self):
+        city = request.form.get('City')
+        unit = request.form.get("Units")
+        sample_data_path = os.path.join(os.getenv("project_root_path"), "webapp/data/sample_data.json")
+        weather_data = json.load(open(sample_data_path))
+        batch_insert_entity_operations = []
+        for w in weather_data:
+            if w["StationName"] == city:
+                entity = {
+                    "PartitionKey": w["StationName"],
+                    "RowKey": "{} {}".format(w["ObservationDate"], w["ObservationTime"]),
+                    "Temperature": round((w["Temperature"] - 32) * 5 / 9, 2) if unit == "US" else w["Temperature"],
+                    "Humidity": w["Humidity"],
+                    "Barometer": round(w["Barometer"] * 33.864, 2) if unit == "US" else w["Barometer"],
+                    "WindDirection": w["WindDirection"],
+                    "WindSpeed": round(w["WindSpeed"] * 1.609, 2) if unit == "US" else w["WindSpeed"],
+                    "Precipitation": round(w["Precipitation"] * 25.4, 2) if unit == "US" else w["Precipitation"]
+                }
+                batch_insert_entity_operations.append(("create", entity))
```

Comment:
Could this ever be run twice in a row, in which case using `upsert` might be safer?
Or do we want this to raise an error if they insert the same data twice?

Reply:
I have updated `create` to `upsert`.